### PR TITLE
Fix ray recompilation and accuracy

### PR DESCRIPTION
**Pull request by sixiang-google**

### CODE REVIEW

File: jetstream_pt/ray_worker.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def _call_model_generate(
           )
           for k, v in torchjax.to_torch(caches)
       ]
-    mask = jnp.expand_dims(mask, (1, 2))
+    mask = jnp.expand_dims(new_mask, (1, 2))
```
Comment (FanhaiLu1) [inline jetstream_pt/ray_worker.py:360]: Thanks for caching it!

**Review by FanhaiLu1 — APPROVED** [review]
